### PR TITLE
[Feature][Zeta] Add COS support for checkpoint storage

### LINKED ISSUE #7924: [Improvement][Checkpoint-storage-plugins] checkpoint storage type support COS
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/seatunnel/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

SeaTunnel Engine does not support COS(Tencent Cloud Object Storage) as checkpoint storage type

### Usage Scenario

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by lm-ylj**
### Purpose of this pull request

close #7924

### How was this patch tested?

SeaTunnel.yaml:
```yaml
seatunnel:
  engine:
    history-job-expire-minutes: 1440
    backup-count: 1
    queue-type: blockingqueue
    print-execution-info-interval: 60
    print-job-metrics-info-interval: 60
    slot-service:
      dynamic-slot: true
    checkpoint:
      interval: 10000
      timeout: 60000
      storage:
        type: hdfs
        max-retained: 3
        plugin-config:
          namespace: /seatunnel-test
          storage.type: cos
          cos.bucket: cosn://seatunnel-1251215091
          fs.cosn.credentials.provider: org.apache.hadoop.fs.cosn.auth.SimpleCredentialsProvider
          fs.cosn.userinfo.secretId: xxx
          fs.cosn.userinfo.secretKey: xxx
          fs.cosn.bucket.region: xxx
```

RESTful API: http://192.168.162.200:5801/hazelcast/rest/maps/submit-job
```json
{
    "env": {
      "execution.parallelism": 1,
      "job.mode": "streaming",
      "checkpoint.interval": 10000
    },
    "source": [
      {
        "plugin_name": "MySQL-CDC",
        "base-url": "jdbc:mysql://192.168.162.200:3306/sea_tunnel_test",
        "username": "root",
        "password": "xxxx",
        "table-names": ["sea_tunnel_test.pay_channel"],
        "startup.mode": "initial"
      }
    ],
    "sink": [
        {
            "plugin_name": "Redis",
            "host": "192.168.162.200",
            "port": 6379,
            "auth": "xxxx",
            "key": "pay:pay_channel",
            "data_type": "key",
            "batch_size": 1
        }
    ]
}
```

execute result:
```text
{
    "jobId": "903107743953453057",
    "jobName": "SeaTunnel_Job"
}
```

Tencent COSBrowser:
image

### Check list

* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  2. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
  3. Add ci label in [label-scope-conf](https://github.com/apache/seatunnel/blob/dev/.github/workflows/labeler/label-scope-conf.yml)
  4. Add e2e testcase in [seatunnel-e2e](https://github.com/apache/seatunnel/tree/dev/seatunnel-e2e/seatunnel-connector-v2-e2e/)
  5. Update connector [plugin_config](https://github.com/apache/seatunnel/blob/dev/config/plugin_config)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Comment by lm-ylj** [comment]
The following three jar packages are required in the lib directory:
- hadoop-cos-3.4.1.jar
- cos_api-bundle-5.6.69.jar
- hadoop-shaded-guava-1.1.1.jar

**Comment by Hisoka-X** [comment]
> The following three jar packages are required in the lib directory:
> 
> * hadoop-cos-3.4.1.jar
> * cos_api-bundle-5.6.69.jar
> * hadoop-shaded-guava-1.1.1.jar

Should we tell this to users in doc?

**Comment by lm-ylj** [comment]
> > The following three jar packages are required in the lib directory:
> > 
> > * hadoop-cos-3.4.1.jar
> > * cos_api-bundle-5.6.69.jar
> > * hadoop-shaded-guava-1.1.1.jar
> 
> Should we tell this to users in doc?

Okay，I‘ll write it down in the doc
done

File: docs/en/seatunnel-engine/checkpoint-storage.md [hunk 1]
```diff
@@ -73,6 +73,42 @@ For additional reading on the Hadoop Credential Provider API, you can see: [Cred
...
+```
+
+For additional reading on the Hadoop Credential Provider API, you can see: [Credential Provider API](https://hadoop.apache.org/docs/stable/hadoop-project-dist/hadoop-common/CredentialProviderAPI.html).
+
+For additional COS configuration, you can see: [Tencent Hadoop-COS Docs](https://doc.fincloud.tencent.cn/tcloud/Storage/COS/846365/hadoop)
+
+Please add the following jar to the lib directory:
+- hadoop-cos-3.4.1.jar
+- cos_api-bundle-5.6.69.jar
+- hadoop-shaded-guava-1.1.1.jar
```
Comment (Hisoka-X) [inline docs/en/seatunnel-engine/checkpoint-storage.md]: Please add jar download path of https://mvnrepository.com/. So users can download it by themself.

Comment (lm-ylj) [reply docs/en/seatunnel-engine/checkpoint-storage.md]: done

**Review by Hisoka-X — APPROVED** [review]
LGTM. Thanks @lm-ylj !

**Review by hailin0 — DISMISSED** [review]

**Comment by hailin0** [comment]
waiting for ci passed

**Review by hailin0 — APPROVED** [review]

**Pull request merged**